model/cli/spans.yaml

```diff
@@ -0,0 +1,21 @@
+groups:
+  - id: span.cli.internal
+    type: span
+    span_kind: internal
+    stability: experimental
+    brief: >
+      Describes CLI (Command Line Interfaces) programs span.
```

Comment:
What do you mean with both sides? A caller of a CLI program and a the CLI program itself? Had a chat around this  topics with @AlexanderWert today and given we added `exit.code` as required field probably must say this is not for daemon apps, just for synchronous / foreground ones.

Reply:
> What do you mean with both sides? A caller of a CLI program and a the CLI program itself?

yep, this.

> I think this can work for both cases.

Then perhaps it's worth mentioning this., but also we'd need a some way to differentiate these two spans in some way. I'd argue that the caller span should probably be `client` instead of `internal` .

> probably must say this is not for daemon apps, just for synchronous / foreground ones.

we can say that it's intended for short-lived apps focused on executing one tasks (I don't think days/weeks long spans are useful no matter if they describe daemon or a regular process)

Reply:
> > What do you mean with both sides? A caller of a CLI program and a the CLI program itself?
> 
> yep, this.
> 
> > I think this can work for both cases.
> 
> Then perhaps it's worth mentioning this., but also we'd need a some way to differentiate these two spans in some way. I'd argue that the caller span should probably be `client` instead of `internal` .
> 
> > probably must say this is not for daemon apps, just for synchronous / foreground ones.
> 
> we can say that it's intended for short-lived apps focused on executing one task (I don't think days/weeks long spans are useful no matter if they describe daemon or a regular process)

Updated PR, not sure that duplicating the attributes for the spans kind was the right call.

Reply:
I like it! There are ways to avoid copy-paste, but it's pretty minimal here and it has no effect on produced markdown.

I just left a couple of suggestions to add headings before tables, otherwise looks great!

Reply:
Cool, applied both suggestions, thanks!